resource_customizations/work.karmada.io/ResourceBinding/health.lua

```diff
@@ -0,0 +1,38 @@
+health_status = {}
+if ( obj.status ~= nil and obj.spec ~= nil ) then
+  if obj.spec.clusters ~= nil then
+    if #obj.spec.clusters == 0 then
+      health_status.status = "Progressing"
+      health_status.message = "Current resource status is insufficient"
+      return health_status
+    end
+    if obj.status.aggregatedStatus ~= nil then
+      if #obj.spec.clusters ~= #obj.status.aggregatedStatus then
+        health_status.status = "Progressing"
+        health_status.message = "Current resource status is insufficient"
+        return health_status
+      end
+      for i, status in ipairs(obj.status.aggregatedStatus) do
+        if status.health ~= nil then
+          if status.health == "Unhealthy" then
+            health_status.status = "Degraded"
+            health_status.message = "Current resource status is unhealthy"
+            return health_status
+          end
+          if status.health == "Unknown" then
+            if ( status.applied ~= nil and status.applied == false ) then
```

Comment:
reach here, there will be 7 nest layers, which might be not good for readability.